QuoridorEngine/src/core/QuoridorGame.cs

```diff
@@ -261,7 +276,22 @@ public void UndoMoves(int x)
             if(x <= 0 || x > gameHistory.Count) throw new ArgumentException();
 
             for(int i = 0; i < x; i++)
+            {
                 UndoMove(gameHistory.Pop());
+                boardZobristHashes.Pop();
+            }    
+        }
+
+        /// <summary>
+        /// Get the zobrist hash of to reference the current game state in a transposition table
+        /// </summary>
+        /// <param name="isWhitePlayer">Player whose turn it is</param>
+        /// <returns>Current game state's zobrist hash</returns>
+        public long GetZobristHash(bool isWhitePlayer)
```

Comment:
Why should it matter to the hashing function whose turn it is? This is not information the QuoridorState class should know